### PR TITLE
fix(command-link): don't modify gitignore if a pattern includes .netlify

### PR DESCRIPTION
**Pull request by erezrokah**
**- Summary**

Partially fixes https://github.com/netlify/cli/issues/2029 (I'll open other separates issue per https://github.com/netlify/cli/issues/2029#issuecomment-803581401)

**- Test plan**

On a linked repo change your `.gitignore` `.netlify` pattern to `/.netlify` and re-run `ntl link`.

**- A picture of a cute animal (not mandatory but encouraged)**
image

### CODE REVIEW

File: src/utils/gitignore.js [hunk 1]
```diff
@@ -92,7 +92,7 @@ const ensureNetlifyIgnore = async function (dir) {
     // ignore
   }
   /* Not ignoring .netlify folder. Add to .gitignore */
-  if (!ignorePatterns || !ignorePatterns.patterns.includes('.netlify')) {
+  if (!ignorePatterns || !ignorePatterns.patterns.some((pattern) => pattern.includes('.netlify'))) {
```
Comment (eduardoboucas) [inline src/utils/gitignore.js:95]: Do you think there's a risk that someone uses sub-directories like `mysite.netlify.app`, which we'd be targeting with this new logic and incorrectly skipping the `.gitignore` modification? It's probably an edge case, but I wonder if a regular expression like `(^|\/)\.netlify($|\/)` would be more accurate?

Comment (erezrokah) [reply src/utils/gitignore.js]: Great point. Done in https://github.com/netlify/cli/pull/2031/commits/38497c1640b31f87f7380551b07ea5791e2386ce

Comment (eduardoboucas) [reply src/utils/gitignore.js]: Sorry, I just realised that this might not account for Windows paths? Perhaps `(^|\/|\\)\.netlify($|\/|\\)` is better?

Comment (erezrokah) [reply src/utils/gitignore.js]: Nice one, done in https://github.com/netlify/cli/pull/2031/commits/cff8f941ba917fc8ba3c514d71d5e18d2af6bd1d

**Review by eduardoboucas — APPROVED** [review]
🚀

**Pull request merged**